### ISSUE TITLE
Radio button in PortableText annotation causes popups to disappear in Safari

### ISSUE THREAD

**Issue by nring** [sanity-io/sanity#7351]
**Describe the bug**

I have a `customLink` schema that allows a user to select a radio button to determine if a link is internal or external. Making the selection will show the appropriate field to select the link destination. For the purposes of this bug report, I have left that functionality out and all fields are currently displayed. 

The bug occurs when selecting text in my PortableText editor and then clicking my link annotation component. The customLink component details pop up, but after selecting either radio button option the popup disappears. Sometimes I'm able to re-select the link text and fill in the rest of the details but other times I'm not able to get the popup to appear. 

This only occurs in Safari 17.5 and Technology Preview 18.0. This does not occur in Chrome or Firefox.

**To Reproduce**

1. Go to PortableText component and activate text area.
1. Highlight text
2. Select customLink annotation
3. customLink details pops up
4. Select a radio button
5. customLink details immediately disappear

**Expected behavior**

I would expect the `customLink` details popup to stay open after selecting an Internal or External link.

**Screenshots**

https://github.com/user-attachments/assets/3ce4e576-8dbe-4df3-a73b-b268ec05c11b

**Which versions of Sanity are you using?**
```
@sanity/cli (global)          3.52.4 (latest: 3.53.0)
@sanity/cli                   3.53.0 (up to date)
@sanity/dashboard              3.1.6 (latest: 4.0.0)
@sanity/eslint-config-studio   4.0.0 (up to date)
@sanity/icons                  3.3.1 (up to date)
@sanity/vision                3.53.0 (up to date)
sanity                        3.53.0 (up to date)
```

**What operating system are you using?**
MacOS Sonoma. 

**Which versions of Node.js / npm are you running?**
```
10.7.0
v20.15.1
1.22.19 (yarn)
```

**Additional context**

I recently upgraded to Sanity Studio 3 and this is the only bug I've found so far.

**Comment by nring** [comment]
Confirming this is still an issue after Sanity 3.57.4, which looks to have addressed a number of PortableText decorator and annotation issues.

**Comment by nring** [comment]
This seems to be resolved for me with Sanity Studio 3.68.0. Most likely with one of these (or a combination) of these bug fixes:
```
- Keeps tab focus inside popover dialogs for the Portable Text Editor when editing links and similar.
- Fixes an issue where WebKit was unable to tab into a form field's action menu.
```
